app/src/main/java/bg/zahov/app/ui/history/HistoryScreen.kt

```diff
@@ -173,9 +173,14 @@ fun Workout(item: HistoryWorkout, onItemClick: (String) -> Unit) {
             )
         }
 
-        for (i in 0 until item.exercises.size) {
-            ExerciseWithSets(exerciseName = item.exercises[i], bestSet = item.bestSets[i])
-        }
+        Exercises(exercises = item.exercises, bestSets = item.bestSets)
+    }
+}
+
+@Composable
+fun Exercises(exercises: List<String>, bestSets: List<String>) {
```

Comment:
Why does this need to be a separate function?

Reply:
I wanted to make this reusable if I ever needed to use just the Exercises composable within the Workout.

Reply:
I think it's best to add abstraction only when needed. 
That leaves you with less code to maintain, which is a rule of thumb. The idea behind functions is to avoid using the same block of code more than once, again to leave you with less code to maintain.